source/models/row.py

```diff
@@ -0,0 +1,60 @@
+import os, json, uuid
+from commands_functions.schema_keys import Keys
+from outputs.exceptions import *
+from models.index import Index
+
+class Row:
+    def __init__(self, table, value = None, primary_key = None):
+        self.table = table
+        if value is not None:
+            self.__init_by_value(value)
+        elif primary_key is not None:
+            self.__init_by_primary_key()
+        else:
+            raise WrongInput(message = "the value and PK of the row is null")
+        
+    def __init_by_value(self, value):
+        self.value = value
+        self.primary_key = value[self.table.table_metadata.primary_key]
+
+    def __init_by_primary_key(self,primary_key):
+        self.primary_key = primary_key
+        self.__get_by_primary_key()
+
+    def serialize(self):
+        self.__colomns_name_validate()
+        if self.table.table_metadata.primary_key not in self.value : 
+            self.value[self.table.table_metadata.primary_key] = str(uuid.uuid4().node)
+        row_json_data = json.dumps(self.value, indent = 2)
+        with open(self.get_path(), 'x') as row_file: 
+            row_file.write(row_json_data)
+        self.update_index()
+
+    def row_exists(self):
+        return os.path.isfile(self.get_path())
+
+    def get_path(self):
```

Comment:
All rows should be in folder data

in table folder there is two folders : -
Lock
Data

Reply:
done

Reply:
why u didn't solve this ?